### DISCUSSION TITLE
Elapsed time of response in logs

### DISCUSSION THREAD

**Discussion by santasdi** [encode/httpx#1965] [Q&A] [answered]
Hi there!
I have some problems with response elapsed time in logs.
I use event hooks for logging after each response.

```python

logger = logging.getLogger("http")

def my_log(response):
    extra = {
     "response_time": response.elapsed.seconds()
    }
    logger.info("Get response", extra=extra)


client = httpx.Client(event_hooks={"response": [my_log]})
```
But after update **httpx** version from 0.18.2 to 0.19.0 event hooks are now also called for any additional redirect or auth requests.So I catch exception like this _"'.elapsed' may only be accessed after the response has been read or closed."_

How can I collect or calculate elapsed time of response now?
Thanks!

**Comment by lovelydinosaur** [comment, marked as answer]
You can use `response.read()` inside your event handler.

You *won't* properly be supporting streaming responses at that point, but if you wanted to track the total elapsed time in an event hook that makes sense, because the response isn't complete until you've fully downloaded it.
